third_party/katana/lib/usdKatana/readPoints.cpp

```diff
@@ -34,6 +34,49 @@ PXR_NAMESPACE_OPEN_SCOPE
 
 
 static FnKat::Attribute
+_GetPositionAttr(
+        const UsdGeomPoints& points,
+        const PxrUsdKatanaUsdInPrivateData& data)
+{
+    // Because of the logic used to gather these (in PxrUsdInOp::InitUsdInArgs),
+    // we can use the size of this vector as a simple test for whether motion
+    // blur is enabled, without needing access to the cook interface.
+    const std::vector<double>& motionSampleTimes =
+        data.GetUsdInArgs()->GetMotionSampleTimes();
```

Comment:
OK, I think it's actually pretty simple, given how you've structured the computation so nicely to take 'baseTime' as a separate input to the sample times.
```
const std::vector<double> motionSampleTimes = data.GetMotionSampleTimes(points.GetPointsAttr());

// Instead of consulting shutterOpen/shutterClose to build up sampleTimes, iterate through
// motionSampleTImes, adding currentTime to each element, since the fn returns frame-relative
// times
```

I plan to lobby to get this moved up to applying to all PointBased prims in Katana, but probably OK to leave it just on Points to start.